api/src/test/java/org/openmrs/validator/VisitValidatorTest.java

```diff
@@ -448,4 +455,78 @@ public void validate_shouldFailValidationIfFieldLengthsAreNotCorrect() {
 		new VisitValidator().validate(visit, errors);
 		assertEquals(true, errors.hasFieldErrors("voidReason"));
 	}
+	
+	@Test
+	public void validate_shouldFailValidationIfVisitDateBeforeBirthDate() {
+		Visit visit = new Visit();
+		Patient patient = new Patient();
+		calendar.set(1974, 4, 8);
+		patient.setBirthdate(calendar.getTime());
+		patient.setBirthdateEstimated(false);
+		visit.setPatient(patient);
+		calendar.set(1974, 4, 7);
+		visit.setStartDatetime(calendar.getTime());
+		Errors errors = new BindException(visit, "visit");
+		
+		new VisitValidator().validate(visit, errors);
+		
+		assertThat(errors, hasFieldErrors("startDatetime", "Visit.startDateCannotFallBeforeTheBirthDateOfTheSamePatient"));
+	}
+	
+	@Test
+	public void validate_shouldAcceptValidationIfVisitDateWithinEstimatedBirthDateMarginOfError() {
+		Visit visit = new Visit();
+		Patient patient = new Patient();
+		calendar.set(2000, 7, 25);
+		patient.setBirthdate(calendar.getTime());
+		patient.setBirthdateEstimated(true);
+		calendar.set(2010, 7, 25);
+		patient.setDeathDate(calendar.getTime());
```

Comment:
just to understand, we need the deathdate so the age will stay fixed?

Reply:
We do if the birth date is estimated. The reason for this is because the permissive calculation for it says that the visit start date is still valid within half the patients estimated age. So in this test for example, they were born in 2000, died 2010, age 10, so a buffer of 5 years. We can then check that this passes within exactly 5 years and fails exactly outside of 5 years. By contrast if the patient isn't dead their age will continue to grow when the tests are run in the future and as a result the edge for the edge cases will be moving.